### ISSUE TITLE
Moving project panels results in 500 `GetProjectBoard: project board does not exist`

### ISSUE THREAD

**Issue by silverwind** [go-gitea/gitea#29853]
### Description

1. Create Project
2. Create two panels in it
3. Drag-and-drop the panels, swapping them. Observe 500 error from `PUT` request:

```
...web/repo/projects.go:526:checkProjectBoardChangePermissions() [E] GetProjectBoard: project board does not exist [id: 0]
...eb/routing/logger.go:102:func1() [I] router: completed PUT /silverwind/symlink-test/projects/1/0 for 127.0.0.1:56716, 500 Internal Server Error in 30.2ms @ repo/projects.go:546(repo.EditProjectBoard)
```

Reproduced on https://try.gitea.io/silverwind/symlink-test/projects/3308.

### Gitea Version

main

**Comment by lunny** [comment]
I cannot reproduce it on my local main branch.

**Comment by lng2020** [comment]
@lunny It only happens when sorting the default project board

**Comment by lunny** [comment]
This will be resolved by #29874 but that cannot be backport to v1.21. Should we have another PR to do a quick fix for v1.21?

**Comment by silverwind** [comment]
It's not a critical thing, so I would not backport.
